### PR TITLE
UIDATIMP-1440, UIDATIMP-1439, UIDATIMP-1438: Multiple holdings/items

### PR DESCRIPTION
**Pull request by mariia-aloshyna**
This PR is a combination of 3 PRs:
- [UIDATIMP-1440](https://github.com/folio-org/ui-data-import/pull/1417)
- [UIDATIMP-1438](https://github.com/folio-org/ui-data-import/pull/1414)
- [UIDATIMP-1439](https://github.com/folio-org/ui-data-import/pull/1415)

## UIDATIMP-1440
To update the Data Import log JSON tab for items, to accommodate multiple items being created and/or updated and/or errored by importing individual MARC Bibliographic records, without adding more tabs to the JSON screen.

## UIDATIMP-1438
To update the Data Import UI to accommodate multiple holdings and items being created and/or updated by importing individual MARC Bibliographic records

## UIDATIMP-1439
To update the Data Import log JSON tab for holdings, to accommodate multiple holdings being created and/or updated and/or errored by importing individual MARC Bibliographic records, without adding more tabs to the JSON screen

### CODE REVIEW

File: src/routes/JobSummary/components/cells/HoldingsCell.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  holdingsInfo,
+  itemInfo,
+  locations,
+}) => {
+  const holdingsInfoCell = holdingsInfo?.map((holdings, index) => {
+    const entityLabel = getRecordActionStatusLabel(holdings.actionStatus);
+    const holdingsId = holdings.id;
+    const path = `/inventory/view/${instanceId}/${holdingsId}`;
+    const locationId = holdings.permanentLocationId;
+    const isDiscarded = holdings.actionStatus === 'DISCARDED';
```
Comment (mariia-aloshyna) [inline src/routes/JobSummary/components/cells/HoldingsCell.js:23]: @OleksandrHladchenko1 
```suggestion
    const isDiscarded = holdings.actionStatus === RECORD_ACTION_STATUS.DISCARDED;
```

File: src/routes/JobSummary/components/cells/ItemCell.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  sortedItemData,
+  instanceId,
+}) => {
+  const itemsInfoCell = sortedItemData?.map((sortedItems, itemIndex) => {
+    const groupOfItems = sortedItems?.map((item, index) => {
+      const entityLabel = getRecordActionStatusLabel(item.actionStatus);
+      const holdingsId = item.holdingsId;
+      const itemId = item.id;
+      const path = `/inventory/view/${instanceId}/${holdingsId}/${itemId}`;
+      const isDiscarded = item.actionStatus === 'DISCARDED';
```
Comment (mariia-aloshyna) [inline src/routes/JobSummary/components/cells/ItemCell.js:21]: @OleksandrHladchenko1 
```suggestion
      const isDiscarded = item.actionStatus === RECORD_ACTION_STATUS.DISCARDED;
```

File: src/routes/JobSummary/components/RecordsTable.js [hunk 3]
```diff
@@ -117,142 +112,243 @@ export const RecordsTable = ({
...
+        return (
+          <BaseLineCell>
+            <NoValue />
+          </BaseLineCell>
+        );
+      }
+
+      if (isEmpty(itemData)) {
+        const emptyValues = holdingsData?.map((_, index) => (
+          <div key={index} style={{ paddingBottom: '7px' }}>
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.js:219]: Could you please investigate should we move to component `<div key={index} style={{ paddingBottom: '7px' }}>` or `paddingBottom: '7px' ` to css class and reuse across all pull request?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 4]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
-  },
-  relatedHoldingsInfo: {
-    actionStatus: 'CREATED',
-    idList: [faker.random.uuid()],
-  },
-  relatedItemInfo: {
-    actionStatus: 'CREATED',
-    idList: [faker.random.uuid()],
-  },
+  jobExecutionId : '467d7627-c0db-4fb7-b333-4b4983dbf781',
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:49]: Could you please change from ` :` to `:` across all file?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 5]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  sourceRecordOrder : 0,
+  sourceRecordTitle : 'Test record 1',
+  sourceRecordActionStatus : 'CREATED',
+  instanceActionStatus: 'CREATED',
+  error : '',
+  relatedInstanceInfo : {
+    actionStatus : 'CREATED',
+    idList : ['720031b9-a792-4936-963c-a7b63fb96574'],
+    hridList : ['in00000000014'],
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:60]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 6]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  },
+  relatedHoldingsInfo : [],
+  relatedItemInfo : [],
   relatedAuthorityInfo: {
-    actionStatus: 'CREATED',
-    idList: [authorityId],
+    idList : [],
+    hridList : []
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:66]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 7]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+    idList : ['720031b9-a792-4936-963c-a7b63fb96574'],
+    hridList : ['in00000000014'],
+    error : ''
+  },
+  relatedHoldingsInfo : [{
+    actionStatus : 'CREATED',
+    id : 'f648c370-d9d6-432c-a502-b8eb718f867c',
+    permanentLocationId: '53cf956f-c1df-410b-8bea-27f712cca7c0',
+    hrid : 'ho00000000017',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:96]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 8]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  sourceRecordOrder : 4,
+  sourceRecordTitle : 'Test record 2',
+  sourceRecordActionStatus : 'CREATED',
+  instanceActionStatus: 'CREATED',
+  error : '',
+  relatedInstanceInfo : {
+    actionStatus : 'CREATED',
+    idList : ['720031b9-a792-4936-963c-a7b63fb96574'],
+    hridList : ['in00000000014'],
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:89]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 9]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
-    idList: [faker.random.uuid()],
-  },
-  relatedItemInfo: {
-    actionStatus: 'DISCARDED',
-    idList: [faker.random.uuid()],
-  },
+    id : '5cadf17f-eb72-475c-a2e0-7e56f54f0000',
+    permanentLocationId: '758258bc-ecc1-41b8-abca-f7b610822ffd',
+    hrid : 'ho00000000018',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:102]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 10]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  }, {
+    actionStatus : 'DISCARDED',
+    id : '5cadf17f-eb72-475c-a2e0-7e56f54fd3b4',
+    permanentLocationId: 'fcd64ce1-6995-48f0-840e-89ffa2288371',
+    hrid : 'ho00000000014',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:108]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 11]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  }],
+  relatedItemInfo : [{
+    actionStatus : 'CREATED',
+    id : '37f20cb1-f60c-4195-80a1-00a16c4af5cb',
+    hrid : 'it00000000015',
+    holdingsId : 'f648c370-d9d6-432c-a502-b8eb718f867c',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:115]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 12]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  }, {
+    actionStatus : 'DISCARDED',
+    id : 'ccd19bf0-add1-46bb-899b-c457fd448b51',
+    hrid : 'it00000000016',
+    holdingsId : 'f648c370-d9d6-432c-a502-b8eb718f867c',
+    error : 'test error'
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:121]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 13]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  }, {
+    actionStatus : 'CREATED',
+    id : 'ccd19bf0-add1-46bb-899b-c457fd441111',
+    hrid : 'it00000000019',
+    holdingsId : '5cadf17f-eb72-475c-a2e0-7e56f54f0000',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:127]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 14]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  }, {
+    actionStatus : 'CREATED',
+    id : '3bcfe427-a747-405f-b3f3-1d842ffb2222',
+    hrid : 'it00000000018',
+    holdingsId : '5cadf17f-eb72-475c-a2e0-7e56f54f0000',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:133]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 15]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  }, {
+    actionStatus : 'CREATED',
+    id : '3bcfe427-a747-405f-b3f3-1d842ffb49c6',
+    hrid : 'it00000000014',
+    holdingsId : 'f648c370-d9d6-432c-a502-b8eb718f867c',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:139]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 16]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
+  }],
   relatedAuthorityInfo: {
-    actionStatus: 'DISCARDED',
-    idList: [faker.random.uuid()],
+    idList : [],
+    hridList : []
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:143]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 17]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
-  },
-}, {
-  sourceRecordId: sourceRecordsIds[6],
-  sourceRecordOrder: '6',
-  sourceRecordTitle: 'Test item 7',
-  relatedInstanceInfo: {
-    actionStatus: 'DISCARDED',
-    idList: [faker.random.uuid()],
+    idList: [],
+    hridList : []
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:147]: Could you please add comma?

File: src/routes/JobSummary/components/RecordsTable.test.js [hunk 18]
```diff
@@ -20,218 +19,171 @@ window.open.mockReturnValue({ focus: jest.fn() });
...
-  relatedAuthorityInfo: {
-    actionStatus: 'DISCARDED',
-    idList: [faker.random.uuid()],
-  },
-  relatedPoLineInfo: {
-    actionStatus: 'CREATED',
-    idList: [faker.random.uuid()],
+  relatedInvoiceInfo : {
+    idList : [],
+    hridList : []
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/RecordsTable.test.js:151]: Could you please add comma?

File: src/routes/JobSummary/components/cells/tests/InstanceCell.test.js [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+
+describe('InstanceCell component', () => {
+  it('should be rendered with no axe errors', async () => {
+    const { container } = renderInstanceCell('CREATED');
+
+    await runAxeTest({ rootNode: container });
+  });
+
+  describe('when instance was created', () => {
+    it('hotlink should be rendered', () => {
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/cells/tests/InstanceCell.test.js:46]: Could you please rename to  `should render hotlink`?

File: src/routes/JobSummary/components/cells/tests/InstanceCell.test.js [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+      const { container } = renderInstanceCell('CREATED');
+      const instanceHotlink = container.querySelector('[data-test-entity-name="instance"]');
+
+      expect(instanceHotlink.href).toContain('/inventory/view/testInstanceId');
+    });
+  });
+
+  describe('when instance was updated', () => {
+    it('hotlink should be rendered', () => {
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/cells/tests/InstanceCell.test.js:55]: Could you please rename to  `should render hotlink`?

File: src/routes/JobSummary/components/cells/tests/OrderCell.test.js [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+  relatedInstanceInfo: {
+    actionStatus: 'CREATED',
+    idList: [faker.random.uuid()],
+  },
+  relatedHoldingsInfo: [{
+    actionStatus : 'CREATED',
+    id : 'testHoldingsId',
+    permanentLocationId: faker.random.uuid(),
+    hrid : 'ho00000000017',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/cells/tests/OrderCell.test.js:28]: Could you please add comma?

File: src/routes/JobSummary/components/cells/tests/OrderCell.test.js [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+  }],
+  relatedItemInfo: [{
+    actionStatus : 'CREATED',
+    id : '37f20cb1-f60c-4195-80a1-00a16c4af5cb',
+    hrid : 'it00000000015',
+    holdingsId : 'testHoldingsId',
+    error : ''
```
Comment (Dmitriy-Litvinenko) [inline src/routes/JobSummary/components/cells/tests/OrderCell.test.js:35]: Could you please add comma?

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Pull request merged**